### PR TITLE
PUBDEV-3877: Implement MOJO for Stacked Ensembles

### PR DESCRIPTION
**Pull request by navdeep-G**
* This PR adds MOJO support for Stacked Ensembles

### CODE REVIEW

**Comment by navdeep-G** [comment]
Will be tweaking a bit, but wanted this PR open first.

**Comment by navdeep-G** [comment]
Should be ready for review now @mmalohlava @michalkurka @ledell

**Comment by ledell** [comment]
@navdeep-G This looks good to me.  The tests check all three cases (regression, binomial, multiclass) so I think we are covering everything here.  Is there anything else we should be testing for with MOJOs?

**Comment by michalkurka** [comment]
"Files changed 1,386"
- is the majority of the files the binary representations of the trees?

If that is the case we need to lower this number. Smaller models would be do as well for the unit tests + we can consider using a ZIP instead of the unpacked files. It is too much right now

File: h2o-genmodel/src/main/java/hex/genmodel/algos/ensemble/StackedEnsembleMojoModel.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    MojoModel[] _baseModels; //An array of base models
+    int _baseModelNum; //Number of base models
+
+    public StackedEnsembleMojoModel(String[] columns, String[][] domains, String responseColumn) {
+        super(columns, domains, responseColumn);
+    }
+
+    @Override
+    public double[] score0(double[] row, double[] preds) {
+        double[] basePreds = new double[_baseModelNum]; //Proper allocation for binomial and regression ensemble (one prediction per base model)
```
Comment (michalkurka) [inline h2o-genmodel/src/main/java/hex/genmodel/algos/ensemble/StackedEnsembleMojoModel.java]: Allocating memory per-row is not ideal however I understand there are no good solutions here

File: h2o-genmodel/src/test/java/hex/genmodel/algos/ensemble/StackedEnsembleMultinomialMojoTest.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            put("PSA", "1.4");
+            put("VOL", "0");
+            put("GLEASON", "6");
+        }});
+
+        assertEquals(1, pred.labelIndex);
+        assertEquals("1", pred.label);
+        assertArrayEquals(new double[]{0.006592327, 0.901237, 0.09217069}, pred.classProbabilities, 1e-5);
+
+        System.out.println("Has RACE: " + pred.label);
```
Comment (michalkurka) [inline h2o-genmodel/src/test/java/hex/genmodel/algos/ensemble/StackedEnsembleMultinomialMojoTest.java]: What is the purpose of the print statements?

File: h2o-genmodel/src/test/java/hex/genmodel/algos/ensemble/StackedEnsembleRegressionMojoTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            put("DPROS", "2");
+            put("DCAPS", "1");
+            put("PSA", "1.4");
+            put("VOL", "0");
+            put("GLEASON", "6");
+        }});
+
+        assertEquals(66.29695, pred.value, 1e-5);
+
+        System.out.println("Predicted AGE: " + pred.value);
```
Comment (michalkurka) [inline h2o-genmodel/src/test/java/hex/genmodel/algos/ensemble/StackedEnsembleRegressionMojoTest.java]: Do we need this? We should know what the value is - we have the assert.

File: h2o-genmodel/src/test/resources/hex/genmodel/algos/ensemble/regression/models/GBM/GBM_model_R_1510186395165_2947/experimental/README.md [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Outputting model information in JSON is an experimental feature and we appreciate any feedback.
```
Comment (michalkurka) [inline h2o-genmodel/src/test/resources/hex/genmodel/algos/ensemble/regression/models/GBM/GBM_model_R_1510186395165_2947/experimental/README.md]: I don't think we need the "experimental" directories in Git.

File: h2o-genmodel/src/main/java/hex/genmodel/algos/ensemble/StackedEnsembleMojoModel.java (hunk 1 above)
Comment (navdeep-G) [reply h2o-genmodel/src/main/java/hex/genmodel/algos/ensemble/StackedEnsembleMojoModel.java]: Yes, I could not think of a better way given the current API. But worth modifying in the short future, if possible.

File: h2o-genmodel/src/test/java/hex/genmodel/algos/ensemble/StackedEnsembleMultinomialMojoTest.java (hunk 2 above)
Comment (navdeep-G) [reply h2o-genmodel/src/test/java/hex/genmodel/algos/ensemble/StackedEnsembleMultinomialMojoTest.java]: No purpose. I can remove.

File: h2o-genmodel/src/test/java/hex/genmodel/algos/ensemble/StackedEnsembleRegressionMojoTest.java (hunk 3 above)
Comment (navdeep-G) [reply h2o-genmodel/src/test/java/hex/genmodel/algos/ensemble/StackedEnsembleRegressionMojoTest.java]: Nope, I can remove.

File: h2o-genmodel/src/test/resources/hex/genmodel/algos/ensemble/regression/models/GBM/GBM_model_R_1510186395165_2947/experimental/README.md (hunk 4 above)
Comment (navdeep-G) [reply h2o-genmodel/src/test/resources/hex/genmodel/algos/ensemble/regression/models/GBM/GBM_model_R_1510186395165_2947/experimental/README.md]: Correct, I can remove.

**Comment by navdeep-G** [comment]
@michalkurka Yes, those are all of the binary rep of trees. What I can do is remove the "experimental" directory and zip up those files. Is that okay?

**Comment by michalkurka** [comment]
I think having it in the zip file is okay.

The expanded MOJOs are good for something like GLM but no so much for GBM.

**Comment by navdeep-G** [comment]
@michalkurka I have implemented your suggestions. Please let me know if anything else is needed.

**Pull request merged**